mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingBatchCoordinator.java

```diff
@@ -76,6 +79,15 @@ public PojoMassIndexingBatchCoordinator(PojoMassIndexingMappingContext mappingCo
 		this.dropAndCreateSchemaOnStart = dropAndCreateSchemaOnStart;
 		this.purgeAtStart = purgeAtStart;
 		this.mergeSegmentsAfterPurge = mergeSegmentsAfterPurge;
+
+		this.agentStartContext = new PojoMassIndexerAgentStartContextImpl(
+				mappingContext.threadPoolProvider()
+						.newScheduledExecutor(
+								1,
+								PojoMassIndexingBatchIndexingWorkspace.THREAD_NAME_PREFIX + "Mass indexer agent"
+						),
```

Comment:
We don't always need this executor, though... only when using outbox-polling coordination.

Can you create the executor lazily, on the first call to `PojoMassIndexerAgentStartContextImpl#scheduledExecutor`? That way we don't use resources unnecessarily.

Reply:
I went with adding `clear()` to the start context, as otherwise, we'd call `agentStartContext.scheduledExecutor().shutdownNow()` and start/close the executor which is probably not what we'd want.